test-data/unit/plugins/add_classmethod.py

```diff
@@ -0,0 +1,26 @@
+from mypy.plugin import Plugin
+from mypy.nodes import ARG_POS, Argument, Var
+from mypy.plugins.common import add_method
+from mypy.types import NoneType
+
+
+class ClassMethodPlugin(Plugin):
+    def get_base_class_hook(self, fullname: str):
+        if "BaseAddMethod" in fullname:
+            return add_extra_methods_hook
+        return None
+
+
+def add_extra_methods_hook(ctx):
```

Comment:
Can you please type all arguments / returns? This is just a style we follow in other test plugins.